rust/ip-packet/src/lib.rs

```diff
@@ -21,13 +21,21 @@ use icmpv4_header_slice_mut::Icmpv4HeaderSliceMut;
 use icmpv6_header_slice_mut::Icmpv6EchoHeaderSliceMut;
 use ipv4_header_slice_mut::Ipv4HeaderSliceMut;
 use ipv6_header_slice_mut::Ipv6HeaderSliceMut;
-use std::{
-    net::{IpAddr, Ipv4Addr, Ipv6Addr},
-    ops::{Deref, DerefMut},
-};
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 use tcp_header_slice_mut::TcpHeaderSliceMut;
 use udp_header_slice_mut::UdpHeaderSliceMut;
 
+pub const MAX_IP_SIZE: usize = MTU + WG_OVERHEAD + NAT46_OVERHEAD + DATA_CHANNEL_OVERHEAD;
+
+pub const MTU: usize = 1280;
+
+/// Wireguard has a 32-byte overhead (4b message type + 4b receiver idx + 8b packet counter + 16b AEAD tag)
+const WG_OVERHEAD: usize = 32;
+/// In order to do NAT46 without copying, we need 20 extra byte in the buffer (IPv6 packets are 20 byte bigger than IPv4).
+const NAT46_OVERHEAD: usize = 20;
+/// TURN's data channels have a 4 byte overhead.
+const DATA_CHANNEL_OVERHEAD: usize = 4;
```

Comment:
@conectado I had to move these to the `ip_packet` crate so that the rest of the crates have access to it. It makes it slightly less generic because now we "know" about things like WG and TURN overhead. I don't think it is worth it to decouple that though because we aren't going to use `IpPacket` anywhere else for now and if we do, we can always still make it a const-generic.

Reply:
the `MTU` here reads a bit weird, just a naming thing due to the loss of context now being in this crate.

I propose we rename it to `TUNNEL_MTU`.

Reply:
I named it `PACKET_SIZE`.